### PR TITLE
Issue #701 Updated instructions for KVM wrt deleting existing VMs…

### PR DESCRIPTION
**Pull request by Preeticp**
…also changed TOC levels for better visibility of issues within the TOC itself.

@hferentschik, out of the 7 issues in this doc, 5 were specifically to do with the drivers, hence rather than do away with identifying issues with the drivers, I have marked out those issues specific to drivers and those that are generic, this is in line with CDK docs too.

To improve visibility of the individual issues, I have changed the TOC to include level 2 which in OpenShift docs should render similar to this, afaics: https://docs.openshift.org/latest/dev_guide/application_lifecycle/promoting_applications.html.

@LalatenduMohanty pointed out that the Uninstalling section should be a separate document from the Installing one, currently it is not exactly visible and hence easily missed out by users. This section already had the instructions provided by @LalatenduMohanty, I have merely added a link to the KVM section. Let me know your thoughts on separating the Uninstall section into a separate doc.

### CODE REVIEW

File: docs/source/getting-started/installing.adoc [hunk 1]
```diff
@@ -128,4 +128,4 @@ $ rm -rf ~/.kube
 
 .  With your hypervisor management tool, confirm that there are no
 remaining artifacts related to the Minishift VM. For example, if you use
-KVM, you need to run the `virsh` command.
+KVM, you need to run the `virsh` command to avoid the issue link:../using/troubleshooting{outfilesuffix}#domain-minishift-already-exists[Error starting the VM: ...operation failed: domain 'minishift' already exists...].
```
Comment (LalatenduMohanty) [inline docs/source/getting-started/installing.adoc]: @Preeticp I do not think this is required. 914172
Users who want to know uninstall  steps for Minishift should not get redirected to  troubleshooting issue `Error starting the VM: ...operation failed: domain 'minishift' already exists`.

**Comment by LalatenduMohanty** [comment]
Apart from the single comment, rest LGTM.

File: docs/source/getting-started/installing.adoc (hunk 1 above)
Comment (budhrg) [inline docs/source/getting-started/installing.adoc]: Ya agree

**Comment by Preeticp** [comment]
@LalatenduMohanty, removed the link.

**Comment by LalatenduMohanty** [comment]
@Preeticp #722

File: docs/source/using/troubleshooting.adoc [hunk 2]
```diff
@@ -13,11 +13,11 @@ toc::[]
 This section contains solutions to common problems that you might
 encounter while using Minishift.
 
-[[troubleshooting-kvm]]
-== KVM driver
+[[troubleshooting-drivers]]
+== Troubleshooting problems with drivers
```
Comment (hferentschik) [inline docs/source/using/troubleshooting.adoc]: I'd argue that one first has general issues and then driver issues. WDYT?

Also "Troubleshooting" seems redundant for this heading and make is imo unnecessarily long.

File: docs/source/using/troubleshooting.adoc [hunk 3]
```diff
@@ -51,21 +51,39 @@ $ systemctl enable virtlogd
 ----
 
 [[domain-minishift-already-exists]]
-=== Error starting the VM: ... operation failed: domain 'minishift' already exists ...
+=== KVM: `Error starting the VM: ...operation failed: domain 'minishift' already exists...`
```
Comment (hferentschik) [inline docs/source/using/troubleshooting.adoc]: This title is too long. We should just add the essential part of the error message, e.g. 

KVM:  'minishift' already exists...

File: docs/source/using/troubleshooting.adoc [hunk 4]
```diff
@@ -51,21 +51,39 @@ $ systemctl enable virtlogd
...
++
+----
+ $ rm -rf ~/.minishift/machines
+----
+
+In case all of this fails, you may want to link:../getting-started/installing{outfilesuffix}#uninstall-instructions[uninstall Minishift] and do a fresh install of Minishift.
 
 [[create-vmnet-interface-permission]]
-=== Error: could not create vmnet interface, permission denied or no entitlement
+=== xhyve: `Error: could not create vmnet interface, permission denied or no entitlement`
```
Comment (hferentschik) [inline docs/source/using/troubleshooting.adoc]: Also too long, maybe "xhyve: could not create vmnet interface"

File: docs/source/using/troubleshooting.adoc [hunk 5]
```diff
@@ -91,23 +109,21 @@ system.
...
 
-[[troubleshooting-authentication]]
-== Users and authentication
+
+[[troubleshooting-general-problems]]
+== Troubleshooting general problems
 
 [[special-characters-passwords]]
-=== Some special characters cause passwords to fail
+=== Users and authentication: Some special characters cause passwords to fail
```
Comment (hferentschik) [inline docs/source/using/troubleshooting.adoc]: I think 'Users and authentication:' is not required. "special characters cause passwords to fail" should do.

File: docs/source/using/troubleshooting.adoc [hunk 6]
```diff
@@ -116,11 +132,8 @@ cause passwords to fail.
 Workaround: When creating and entering passwords, wrap the string with
 single quotes in the following format: `<password>`
 
-[[troubleshooting-snapshots]]
-== Snapshots with virsh
-
 [[minishift-delete-fails-undefine-snapshots]]
-=== `minishift delete` fails to undefine snapshots of running instances made using virsh on KVM/libvirt
+=== Snapshots with virsh: `minishift delete` fails to undefine snapshots of running instances made using virsh on KVM/libvirt
```
Comment (hferentschik) [inline docs/source/using/troubleshooting.adoc]: "Undefining virsh snapshots fail"

File: docs/source/using/troubleshooting.adoc [hunk 7]
```diff
@@ -51,21 +51,39 @@ $ systemctl enable virtlogd
...
++
+----
 $ sudo virsh undefine minishift
 ----
 
-[[troubleshooting-xhyve]]
-== xhyve driver
+. Delete the `.minishift/machines` directory using:
++
+----
```
Comment (hferentschik) [inline docs/source/using/troubleshooting.adoc]: I'd start to consistently mark the type of the source examples, in this case we should add 

'[source,sh]'

(this applies to all commands in this page)

File: docs/source/using/troubleshooting.adoc (hunk 2 above)
Comment (Preeticp) [inline docs/source/using/troubleshooting.adoc]: I thought of putting general issues before the Driver issues similar to CDK docs, but then retained the existing order considering that drivers are prerequisites and an issue with them will prevent further use of Minishift. Also, the Driver issues seem like showstoppers and there are 5 of them, whereas we only have 1 general issue out of which only 1 is most likely to affect users often. If you think reversing the order is more useful, I will change it.

**Comment by Preeticp** [comment]
@hferentschik rebasing did not help me get the L&F of the final site, maybe I am missing something. One suggestion: do you think it would be better to make the issues like 'KVM: Error creating new host: dial tcp: missing address' as heading level 4 and increase the TOC level to 3?
Despite the changes, the titles still seem long.

**Comment by hferentschik** [comment]
>  rebasing did not help me get the L&F of the final site, 

You also need to get the latest changes from master and then rebase your pull request on top of this. Did you do this? Are you still not able to see the doc changes in their final form? I really think you need to see it.

File: docs/source/using/troubleshooting.adoc [hunk 8]
```diff
@@ -78,6 +108,7 @@ Reset the `minishift`-specific IP database, make sure to remove the
 `minishift` entry section from the `dhcpd_leases` file, and reboot your
 system.
 
+[source,sh]
```
Comment (hferentschik) [inline docs/source/using/troubleshooting.adoc]: this is not shell source, we can remove it from there

**Comment by hferentschik** [comment]
Rebased and merged, thanks

**Pull request closed without merging**